lib/Test/Conch.pm

```diff
@@ -100,22 +110,59 @@ sub new {
     return $self;
 }
 
-sub DESTROY {
-    my $self = shift;
+sub DESTROY ($self) {
 
     # ensure that a new Test::Conch instance creates a brand new Mojo::Pg connection (with a
     # possibly-different dsn) rather than using the old one to a now-dead postgres instance
     Conch::Pg->DESTROY;
 }
 
+=head2 init_db
+
+Sets up the database for testing, using the final schema rather than running migrations.
+No data is added -- you must load all desired fixtures.
+
+Note that the Test::PostgreSQL object must stay in scope for the duration of your tests.
+Returns the Conch::DB object as well when called in list context.
+
+=cut
+
+sub init_db ($class) {
+    my $pgsql = Test::PostgreSQL->new(pg_config => 'client_encoding=UTF-8');
+    die $Test::PostgreSQL::errstr if not $pgsql;
+
+    my $schema = Conch::DB->connect(
+        $pgsql->dsn, 'postgres', '',
+        {
+            # same as from Mojo::Pg->new($uri)->options
+            AutoCommit          => 1,
+            AutoInactiveDestroy => 1,
+            PrintError          => 0,
+            PrintWarn           => 0,
+            RaiseError          => 1,
+        },
+    );
+
+    Test::More::note('initializing database with sql/schema.sql...');
+
+    $schema->storage->dbh_do(sub {
+        my ($storage, $dbh, @args) = @_;
+        $dbh->do('CREATE ROLE conch LOGIN');
+        $dbh->do('CREATE DATABASE conch OWNER conch');
+        $dbh->do(path('sql/schema.sql')->slurp_utf8) or BAIL_OUT("Test SQL load failed in $_");
+        $dbh->do('RESET search_path');  # go back to "$user", public
+    });
+
+    return wantarray ? ($pgsql, $schema) : $pgsql;
+}
+
 =head2 location_is
 
 Stolen from Test::Mojo's examples. I don't know why this isn't just part of the interface!
 
 =cut
 
-sub location_is {
-    my ($t, $value, $desc) = @_;
+sub location_is ($t, $value, $desc = 'location header') {
     $desc ||= "Location: $value";
```

Comment:
You have a default in the signature, this won't ever fire now will it?

Reply:
fixed; thanks.